File: js/langchainv1/streaming.ts
import "dotenv/config";
import { initChatModel } from "langchain";

export async function main() {
  console.log("=== OpenAI Streaming ===");
  const oai = await initChatModel("openai:gpt-5-nano", {
    outputVersion: "v1",
  });

  const oaiStream = await oai.stream("Write a haiku about programming.");
  for await (const chunk of oaiStream) {
    console.log(chunk.content);
  }
  console.log("\n");

  console.log("=== Anthropic Streaming ===");
  const anthro = await initChatModel("anthropic:claude-sonnet-4-20250514", {
    outputVersion: "v1",
  });

  const anthroStream = await anthro.stream("Write a haiku about programming.");
  for await (const chunk of anthroStream) {
    console.log(chunk.content);
  }
  console.log("\n");
}

File: python/langchain_v1/streaming.py
import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(__file__)))

from langchain.chat_models import init_chat_model
from langsmith import traceable


@traceable(name="OpenAI Streaming")
def streaming_example_openai(
    output_version: str = "v1", use_responses_api: bool = False
):
    """Stream LangChain v1 response with OpenAI."""
    model = init_chat_model(
        "gpt-4o",
        model_provider="openai",
        output_version=output_version,
        use_responses_api=use_responses_api,
    )

    print("Starting OpenAI streaming...")
    for chunk in model.stream("Write a short poem about artificial intelligence."):
        print(chunk.content, end="", flush=True)

    print("\n\nOpenAI streaming complete!")
    return {"streaming": "complete"}


@traceable(name="Anthropic Streaming")
def streaming_example_anthropic(output_version: str = "v1"):
    """Stream LangChain v1 response with Anthropic."""
    model = init_chat_model(
        "claude-3-5-sonnet-20241022",
        model_provider="anthropic",
        output_version=output_version,
    )

    print("Starting Anthropic streaming...")
    for chunk in model.stream("Write a short poem about artificial intelligence."):
        print(chunk.content, end="", flush=True)

    print("\n\nAnthropic streaming complete!")
    return {"streaming": "complete"}


@traceable
def main(output_version: str = "v1", use_responses_api: bool = False, run_tree=None):
    if run_tree:
        api_suffix = " + Responses API" if use_responses_api else ""
        run_tree.name = f"LangChain Streaming ({output_version}{api_suffix})"
    print("Running LangChain v1 streaming examples...\n")
    streaming_example_openai(output_version, use_responses_api)
    print("\n" + "=" * 50 + "\n")
    if not use_responses_api:
        streaming_example_anthropic(output_version)
    return {"streaming": "complete"}
